clouds/aws/create_aws_network_resources.py

```diff
@@ -0,0 +1,745 @@
+            subnet_count=subnet_count,
+            vpc_id=vpc_id,
+            subnet_cidr_blocks=subnet_cidr_blocks,
+            availability_zones=availability_zones,
+            site_name=site_name,
+        )
+        internet_gateway_id = create_and_attach_igw(
+            ec2_client=ec2_client, site_name=site_name, vpc_id=vpc_id
+        )
+        alloc_ids = allocate_elastic_ips(
+            ec2_client=ec2_client, subnet_count=subnet_count, site_name=site_name
+        )
+        nat_gateway_ids = create_nat_gateway(
+            ec2_client=ec2_client,
+            site_name=site_name,
+            subnet_count=subnet_count,
+            vpc_output=vpc_output,
+            alloc_ids=alloc_ids,
+            subnet_type="public",
+        )
+        rt_public_id = create_public_route_table(
+            ec2_client=ec2_client, site_name=site_name, vpc_id=vpc_id
+        )
+        rt_private_ids = create_private_route_table(
+            ec2_client=ec2_client,
+            subnet_count=subnet_count,
+            vpc_id=vpc_id,
+            site_name=site_name,
+        )
+        associate_route_ipv4(
+            ec2_client=ec2_client,
+            route_table_id=rt_public_id,
+            internet_gateway_id=internet_gateway_id,
+        )
+        associate_route_ipv6(
+            ec2_client=ec2_client,
+            route_table_id=rt_public_id,
+            internet_gateway_id=internet_gateway_id,
+        )
+        associate_route_to_private_nat(
+            ec2_client=ec2_client,
+            subnet_count=subnet_count,
+            rt_private_ids=rt_private_ids,
+            nat_gateway_ids=nat_gateway_ids,
+        )
+        ec2_client.modify_vpc_endpoint(
+            VpcEndpointId=vpc_output["vpc_endpoint"],
+            AddRouteTableIds=rt_private_ids,
+        )
+        public_route_table_association(
+            ec2_client=ec2_client,
+            subnet_count=subnet_count,
+            route_table_id=rt_public_id,
+            subnet_id=vpc_output["public_subnet_ids"],
+        )
+        private_route_table_association(
+            ec2_client=ec2_client,
+            subnet_count=subnet_count,
+            route_table_id=rt_private_ids,
+            subnet_id=vpc_output["private_subnet_ids"],
+        )
+        return vpc_output
+    except ClientError as client_err:
+        LOGGER.error(client_err)
+        raise
+
+
+def associate_route_to_private_nat(
+    ec2_client, subnet_count, rt_private_ids, nat_gateway_ids
+) -> None:
+    """
+    Send private traffic to NAT
+
+    Args:
+         ec2_client (botocore.client.EC2): aws botocore client
+         subnet_count(int): no. of subnets required
+         rt_private_ids (list): list containing id for private route table
+         nat_gateway_ids (list): list consisting nat gateway id's
+    """
+    for count in range(subnet_count):
+        ec2_client.create_route(
+            RouteTableId=rt_private_ids[count],
+            DestinationCidrBlock="0.0.0.0/0",
```

Comment:
will this always be the value?

Reply:
yes, since it for private NAT, 0.0.0.0/0 will be the default route for any IP4 route that is not in route table, otherwise we need to add specific routes in the route table